src/main/java/com/google/devtools/build/lib/syntax/SkylarkList.java

```diff
@@ -521,15 +522,15 @@ public void set(int index, E value, Location loc, Mutability mutability) throws
       name = "extend",
       doc = "Adds all items to the end of the list.",
       parameters = {
-          @Param(name = "items", type = SkylarkList.class, doc = "Items to add at the end.")
+          @Param(name = "items", type = Object.class, doc = "Items to add at the end.")
       },
       useLocation = true,
       useEnvironment = true
     )
     public Runtime.NoneType extend(
-        SkylarkList<E> items, Location loc, Environment env)
+        Object items, Location loc, Environment env)
         throws EvalException {
-      addAll(items, loc, env.mutability());
+      addAll((Collection<? extends E>) EvalUtils.toCollection(items, loc, env), loc, env.mutability());
```

Comment:
That cast isn't sound, and it should result in an "unchecked" warning. Do you know how (and why) these are suppressed in the Bazel codebase? They should really be treated as errors.

Reply:
I am not sure actually why they are suppressed. I removed the cast anyway.